Created RecipeForm, link to 'add recipe'

mostlybadfly opened the pull request:
Creating a new PR for the recipe views for some visibilty.  I am having difficulty incorporating `Ingredient` and `Instruction` fields.  Would this be done utilizing a formset? I am also realizing i'd need to dynamically add new ingredient and instruction fields, I will have to look into some jquery unless django has a way?

mostlybadfly committed:
Created RecipeForm, linke to 'add recipe'

skain commented:
Let's talk about it tomorrow. Formsets are probably the way you ultimately want to go but I think it might be helpful to do a more straightforward, simple approach first.

mostlybadfly committed:
Added login_required decorator to add and edit functions

mostlybadfly committed:
adding LOGIN_URL to settings.py

mostlybadfly committed:
Added ingredient listing and form.

mostlybadfly committed:
Add Instruction view, form and templates.

mostlybadfly commented:
Added views and templates for ingredients and instructions. Realizing now I need a separate 'add' and 'edit' for each rather than just 'edit' so I'll work on that.  For some reason in my recipe instructions view/template, it renders ingredients.  Do you see any reason why?

Update: Found it in my urls.py, I have `instructions` pointing to `views.ingredients` :joy:

mostlybadfly committed:
Fixed urls and views so Instruction list and edit could render

mostlybadfly committed:
Updated view/template names, added recipe edit view, removed blank=true from integer fields.

mostlybadfly committed:
Added Ingredient and Instruction add and edit views, added links

mostlybadfly commented:
I updated `add.html` to `recipe_edit.html` so it can be used by both "add" and "edit" recipe views.  Added urls/views/templates for adding, editing, and viewing ingredients and instructions. Added various links to navigate within different objects.  I'm having difficulty referring back to the object id from my forms in order to create a "back" or "cancel" button.

skain commented on recipes/admin.py:
```diff
@@ -1,5 +1,6 @@
+models = [Recipe, Ingredient, Instruction]
 
-admin.site.register(Recipe)
+admin.site.register(models)
```
Could just be admin.site.register([Recipe, Ingredient, Instruction])

Commit:
back button strategy 1

mostlybadfly committed:
Added ingredients and instructions to recipe detail, created links to go back on forms

mostlybadfly committed:
Updated ingredient_edit with link to go back.

mostlybadfly committed:
Updated admin.py models.

mostlybadfly committed:
Added some styling to recipe detail.

skain commented on recipes/views.py:
```diff
@@ -39,13 +39,10 @@ def edit(request, recipe_id):
+    ingredients = recipe.ingredients.all()
+    instructions = recipe.instructions.all()
+    return render(request, 'recipes/detail.html',
+                  {'recipe': recipe, 'ingredients': ingredients, 'instructions': instructions})
```
Explicitly getting the ingredients and instructions is not really necessary here. In your template you could simply reference recipe.ingredients.all(), etc. 

Having the ability to do it like this is still good though since sometimes your models don't happen to have all the info you need on them.

skain commented on recipes/templates/recipes/detail.html:
```diff
@@ -1,13 +1,35 @@
+    </ul>
+    <h3>Ingredients:</h3>
+    <ul class="list-unstyled d-inline-block text-left">
+    {% for ingredient in ingredients reversed %}
```
Why reversed?

skain reviewed (approved):
Couple small comments but otherwise looks good.
